bundles/org.openhab.binding.homekit/src/main/java/org/openhab/binding/homekit/internal/handler/HomekitBridgeHandler.java

```diff
@@ -0,0 +1,164 @@
+    }
+
+    @Override
+    public void childHandlerInitialized(ThingHandler childHandler, Thing childThing) {
+        // do nothing
+    }
+
+    @Override
+    public void childHandlerDisposed(ThingHandler childHandler, Thing childThing) {
+        // do nothing
+    }
+
+    @Override
+    protected void onConnectedThingAccessoriesLoaded() {
+        createProperties();
+        getThing().getThings().forEach(bridgedThing -> {
+            if (bridgedThing.getHandler() instanceof HomekitAccessoryHandler accessoryHandler) {
+                accessoryHandler.onConnectedThingAccessoriesLoaded();
+            }
+        });
+        onThingOnline();
+    }
+
+    @Override
+    public void onEvent(String jsonContent) {
+        getThing().getThings().forEach(bridgedThing -> {
+            if (bridgedThing.getHandler() instanceof HomekitAccessoryHandler accessoryHandler) {
+                accessoryHandler.onEvent(jsonContent);
+            }
+        });
+    }
+
+    @Override
+    protected void onThingOnline() {
+        updateStatus(ThingStatus.ONLINE);
+        getThing().getThings().forEach(bridgedThing -> {
+            if (bridgedThing.getHandler() instanceof HomekitAccessoryHandler accessoryHandler) {
+                accessoryHandler.onThingOnline();
+            }
+        });
+        super.onThingOnline();
+        HomekitBridgedAccessoryDiscoveryService discoveryService = bridgedAccessoryDiscoveryService;
+        if (discoveryService != null) {
+            discoveryService.startScan();
+        }
+    }
+
+    public void registerDiscoveryService(HomekitBridgedAccessoryDiscoveryService discoveryService) {
+        bridgedAccessoryDiscoveryService = discoveryService;
+    }
+
+    public void unregisterDiscoveryService() {
+        bridgedAccessoryDiscoveryService = null;
+    }
+
+    @Override
+    protected Map<String, Characteristic> getEventedCharacteristics() {
+        eventedCharacteristics.clear();
+        getThing().getThings().forEach(bridgedThing -> {
+            if (bridgedThing.getHandler() instanceof HomekitAccessoryHandler accessoryHandler) {
+                eventedCharacteristics.putAll(accessoryHandler.getPolledCharacteristics());
+            }
+        });
+        return eventedCharacteristics;
+    }
+
+    @Override
+    protected Map<String, Characteristic> getPolledCharacteristics() {
+        polledCharacteristics.clear();
+        getThing().getThings().forEach(bridgedThing -> {
+            if (bridgedThing.getHandler() instanceof HomekitAccessoryHandler accessoryHandler) {
+                polledCharacteristics.putAll(accessoryHandler.getPolledCharacteristics());
+            }
+        });
+        return polledCharacteristics;
+    }
+
+    @Override
+    protected void initializeNotReadyThings() {
+        notReadyThings.clear();
+        notReadyThings.addAll(getThing().getThings()); // a bridge requires all bridged-accessories to be ready
```

Comment:
Should disabled Things be excluded from this `Set` so they don't cause their bridge to never subscribe to events etc.?

Reply:
If things are disabled, then `getThings()` does not include them.

Reply:
.. but we may need some more logic for the case that a new thing is created after the bridge has come online; related to the channelLinked discussion above..

Reply:
> If things are disabled, then getThings() does not include them.

@jlaur I apologise; this statement was wrong; however I am adding a check.

Reply:
> disabled Things .. excluded from this Set so they don't cause their bridge to never subscribe

Done.

> we may need some more logic for the case that a new thing is created after the bridge has come online

Added.